node/resolvers/profile/index.ts

```diff
@@ -44,19 +45,33 @@ export const mutations = {
 
   subscribeNewsletter: async (
     _: any,
-    { email, isNewsletterOptIn }: SubscribeNewsletterArgs,
+    { email, name, isNewsletterOptIn }: SubscribeNewsletterArgs,
     context: Context
   ) => {
     const profile = context.clients.profile
     const optIn =
       isNewsletterOptIn === undefined || isNewsletterOptIn === true
         ? TRUE
         : FALSE
+
+    const updatedPersonalPreferences: PersonalPreferences = {
+      isNewsletterOptIn: optIn,
+    }
+
+    if (name) {
+      const userProfile = await profile.getProfileInfo({ email, userId: '' })
+
+      const userHasFirstName = Boolean(userProfile.firstName)
+
+      // Prevent overrides of the 'firstName' field.
+      if (!userHasFirstName) {
+        updatedPersonalPreferences['firstName'] = name
```

Comment:
Our linter should be complaining about this:

```suggestion
        updatedPersonalPreferences.firstName = name
```

Reply:
Oh, this app is not using the latest version of our linters :(